ml-conversational-analytic-tool/tests/test_commentAnalysis.py

```diff
@@ -0,0 +1,64 @@
+import unittest
+
+from ..commentAnalysis import CommentAnalyzer
+
+class TestCommentAnalysis(unittest.TestCase):
+
+    def test_CommentAnalyzer_init(self):
+        analyzer = CommentAnalyzer(['test'])
+        self.assertEqual({'test': 0}, analyzer.word_count,)
+
+    def test_analyzeComment(self):
+        analyzer = CommentAnalyzer(['test'])
+        result = analyzer.analyzeComment("This is a test PR")
+        self.assertEqual({'test': 1, 'Sentiment': 0.0, 'Code Blocks': 0}, result)
+
+    def test_analyzeComment_with_empty_comment(self):
+        analyzer = CommentAnalyzer(['empty comment'])
+        result = analyzer.analyzeComment("")
+        self.assertEqual({'empty comment': 0, 'Sentiment': 0.0, 'Code Blocks': 0}, result)
+
+    def test_analyzeComment_positive_comment(self):
+        analyzer = CommentAnalyzer(['positive comment'])
+        result = analyzer.analyzeComment("This patch is really good")
+        self.assertEqual({'positive comment': 0, 'Sentiment': 0.4927, 'Code Blocks': 0}, result)
```

Comment:
I think it would make sense to have a setup function with a mock defined for the `SentimentIntensityAnalyzer` with a mock of the `polarity_scores` functionality. 

something like this
```
sentiment_analyzer = SentimentIntensityAnalyzer()
sentiment_analyzer.polarity_scores = MagicMock(return_value={"compound": 1.0})
```

After that, you can set the mock as the `vader_sentiment` value
```
analyzer = CommentAnalyzer(['test'])
analyzer.vader_sentiment = sentiment_analyzer
```

which will then allow you to specify the sentiment value returned by the `analyzeComment`. In the example above, this would return a sentiment of 1.0 regardless of what comment was passed.

There might be a better way to handle it. In the end, we want to get away from hard-coding the sentiment values in any of the tests

Reply:
you might actually be able to use mock instead of SentimentIntensityAnalyzer() as well

Reply:
Done